File: foodtruck-api/foodtruck-api/src/main/java/com/foodtruck/foodtruckapi/service/ReportsService.java
package com.foodtruck.foodtruckapi.service;

import com.foodtruck.foodtruckapi.dto.response.DailySalesResponse;
import com.foodtruck.foodtruckapi.dto.response.MonthlyExpenseResponse;
import com.foodtruck.foodtruckapi.dto.response.PopularItemResponse;

import java.util.List;

public interface ReportsService {
    List<MonthlyExpenseResponse> getMonthlyExpenses();
    List<PopularItemResponse> getPopularItems();
    List<DailySalesResponse> getDailySales();
}

File: frontend/src/services/reportsService.ts
import api from './api';
import type { PopularItemResponse, MonthlyExpenseResponse, DailySalesResponse } from '../types/reports.types';

export const getPopularItems = async () => {
    const response = await api.get<PopularItemResponse[]>('/reports/popular-items');
    return response.data;
}

export const getMonthlyExpenses = async () => {
    const response = await api.get<MonthlyExpenseResponse[]>('/reports/monthly-expenses');
    return response.data;
}

export const getDailySales = async () => {
    const response = await api.get<DailySalesResponse[]>('/reports/daily-sales');
    return response.data;
}